### PR TITLE
chore(sdk): allow serializing pipeline params with `typing` type hints

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
KFP SDK supports serializing pipeline parameters with the following types:
https://github.com/kubeflow/pipelines/blob/706c105c8ca3977a91b54d74caebe8ffb39f6aa7/sdk/python/kfp/components/_data_passing.py#L107-L116
However, it doesn't recognize type hints using `typing` module. For example:
```
from typing import List
def my_func(args: List[int]):
    print(args)
```
This example results compile time error: `There are no registered serializers for type "typing.List[int]"`, even though we have serializer for `List`.

This change allows recognizing `typing.List[int]` as `List` and `typing.Dict[str, Any]` as `Dict`, etc.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/components_tests/test_data_passing.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from ..components import _data_passing
```
Comment (neuromage) [inline sdk/python/kfp/components_tests/test_data_passing.py:17]: from kfp.components ...

File: sdk/python/kfp/components/_data_passing.py [hunk 2]
```diff
@@ -185,3 +187,23 @@ def serialize_value(value, type_name: str) -> str:
     raise TypeError('There are no registered serializers for type "{}".'.format(
         str(type_name),
     ))
+
+def get_collection_type_name(type_name: str) -> Optional[str]:
```
Comment (neuromage) [inline sdk/python/kfp/components/_data_passing.py:191]: can this be private?

**Review by neuromage — COMMENTED** [review]
/lgtm

File: sdk/python/kfp/components/_data_passing.py (hunk 2 above)
Comment (chensun) [reply sdk/python/kfp/components/_data_passing.py]: Done.

File: sdk/python/kfp/components_tests/test_data_passing.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/components_tests/test_data_passing.py:17]: Unfortunately, our presubmit test setup only copies the sub-folders:
https://github.com/kubeflow/pipelines/blob/706c105c8ca3977a91b54d74caebe8ffb39f6aa7/test/presubmit-tests-sdk.sh#L28

and `kfp/components` isn't set up as independent [namespace package](https://packaging.python.org/guides/packaging-namespace-packages/), so `kfp` wouldn't be recognized here.

As a result, all unit tests under this folder have to use relative imports.

Ideally, we should fix the whole file structures for v1, and makes test files sit next to their source files like 
- data_passing.py
- data_passing_test.py

Comment (neuromage) [reply sdk/python/kfp/components_tests/test_data_passing.py:17]: Ack, thanks for explaining! Let's fix it at some later point down the road.

**Comment by Ark-kun** [comment]
Thank you for improving the type support, Chen.

**Pull request merged**